### PR TITLE
Add pix2pix (PyTorch)

### PR DESCRIPTION
**Pull request by NielsRogge**
This PR adds an example script for pix2pix.

To do:

- [ ] use Accelerate
- [ ] improve argument names

### CODE REVIEW

File: huggan/pytorch/pix2pix/train.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+dataset = load_dataset(opt.dataset)
+splitted_dataset = dataset.map(split_images, batched=True)
+transformed_dataset = splitted_dataset.with_transform(transforms)
+
+splits = transformed_dataset['train'].train_test_split(test_size=0.1)
+train_ds = splits['train']
+val_ds = splits['test']
+
+dataloader = DataLoader(train_ds, shuffle=True, batch_size=opt.batch_size, num_workers=opt.n_cpu)
+val_dataloader = DataLoader(val_ds, batch_size=10, shuffle=True, num_workers=1)
```
[the remark this answers is not in the document]
Comment (nateraw) [reply huggan/pytorch/pix2pix/train.py:149]: Any reason why we shouldn't do the above? Again, idk why but on my macbook I need to have this set to 0 for dev.

**Review by nateraw — APPROVED** [review]
This looks great, thank you! Now that you've converted one from that repo, we should be able to add any others from there that we may want. Super cool 🚀

File: huggan/pytorch/pix2pix/train.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (NielsRogge) [reply huggan/pytorch/pix2pix/train.py:149]: Yeah the original author set it to 1 for the validation set

**Comment by NielsRogge** [comment]
Gonna merge this one and re-fork, as it was set to private previously. Will improve this script in a follow-up PR.
